src/main/java/com/hms_networks/americas/sc/extensions/system/http/SCHttpUtility.java

```diff
@@ -118,50 +127,45 @@ private static String httpRequest(String url, String header, String body, String
     if (httpStatus == HTTPX_CODE_NO_ERROR) {
       responseFileString = FileAccessManager.readFileToString(responseFile.getAbsolutePath());
     } else if (httpStatus == HTTPX_CODE_EWON_ERROR) {
-      boolean responseFileCleaned = responseFile.delete();
       throw new SCHttpEwonException(
           "An Ewon error was encountered while performing an HTTP "
               + method
               + " request to "
               + url
-              + "! Data loss may result. (Temporary Response File Cleaned: "
-              + responseFileCleaned
-              + ")");
+              + "! Data loss may result."
+              + (outputFile == null ? "(Temporary File Cleaned: " + responseFile.delete() : ""));
     } else if (httpStatus == HTTPX_CODE_AUTH_ERROR) {
-      boolean responseFileCleaned = responseFile.delete();
       throw new SCHttpAuthException(
           "An authentication error was encountered while performing an HTTP "
               + method
               + " request to "
               + url
-              + "! Data loss may result. (Temporary Response File Cleaned: "
-              + responseFileCleaned
-              + ")");
+              + "! Data loss may result."
+              + (outputFile == null ? "(Temporary File Cleaned: " + responseFile.delete() : ""));
     } else if (httpStatus == HTTPX_CODE_CONNECTION_ERROR) {
-      boolean responseFileCleaned = responseFile.delete();
       throw new SCHttpConnectionException(
           "A connection error was encountered while performing an HTTP "
               + method
               + " request to "
               + url
-              + "! Data loss may result. (Temporary Response File Cleaned: "
-              + responseFileCleaned
-              + ")");
+              + "! Data loss may result."
+              + (outputFile == null ? "(Temporary File Cleaned: " + responseFile.delete() : ""));
     } else {
-      boolean responseFileCleaned = responseFile.delete();
       throw new SCHttpUnknownException(
           "An unknown error ("
               + httpStatus
               + ") was encountered while performing an HTTP "
               + method
               + " request to "
               + url
-              + "! Data loss may result. (Temporary Response File Cleaned: "
-              + responseFileCleaned
-              + ")");
+              + "! Data loss may result."
+              + (outputFile == null ? "(Temporary File Cleaned: " + responseFile.delete() : ""));
     }
 
-    responseFile.delete();
+    // Delete response file (if temporary)
+    if (outputFile == null) {
+      responseFile.delete();
```

Comment:
does File.delete() throw IOException? that should be caught?

Reply:
.delete() does not throw an IOException. This File.delete() call already existed anyways and is just being disabled under certain circumstances.